### PR TITLE
Add function signatures to abstract base methods

### LINKED ISSUE #146: All python abstract base class functions should have correct function signature
Abstract base class functions for tools do not currently have the final desired function signature. For example, `revolve` does not match the MATLAB version:

https://github.com/Severson-Group/eMach/blob/2d005a3a5aea33833bca150f519d5f037b836398/python/emach/tools/tool_abc/toolabc.py#L39-L41

### PR DESCRIPTION
**Pull request by Bharat-Ramadas**
# Description

This PR addresses issue #146. All abstract class methods now have the expected function signatures

### CODE REVIEW

**Review by elsevers — CHANGES_REQUESTED** [review]
Need to provide type hints in the function signatures.

File: python/emach/tools/tool_abc/toolabc.py [hunk 1]
```diff
@@ -6,36 +6,42 @@
...
+from typing import List
+
+from ...model_obj import Location2D
+
 
 # define abstract base class for ToolBase
 class ToolBase(ABC):
     @abstractmethod
-    def open(self): pass
+    def open(self, filename: str, visible: bool) -> None: pass
```
Comment (elsevers) [inline python/emach/tools/tool_abc/toolabc.py]: Indicate `Any` for return value

File: python/emach/tools/tool_abc/toolabc.py [hunk 2]
```diff
@@ -6,36 +6,42 @@
...
     
     @abstractmethod
     def close(self): pass
 
 # define abstract base class for DrawerBase
 class DrawerBase(ABC):
     @abstractmethod
-    def draw_line(self): pass
+    def draw_line(self, startxy: Location2D, endxy: Location2D) -> int: pass
```
Comment (elsevers) [inline python/emach/tools/tool_abc/toolabc.py]: `draw_line` should return a `TokenDraw` object (need to create `TokenDraw` class)

File: python/emach/tools/tool_abc/toolabc.py [hunk 3]
```diff
@@ -6,36 +6,42 @@
...
     
     @abstractmethod
-    def draw_arc(self): pass
+    def draw_arc(self, centerxy: Location2D, startxy: Location2D,
```
Comment (elsevers) [inline python/emach/tools/tool_abc/toolabc.py]: `draw_arc` should return a `TokenDraw` object (need to create `TokenDraw` class)

File: python/emach/tools/tool_abc/toolabc.py [hunk 4]
```diff
@@ -6,36 +6,42 @@
...
+    def draw_arc(self, centerxy: Location2D, startxy: Location2D, 
+                 endxy: Location2D) -> int: pass
 
     @abstractmethod
     def select(self): pass
 
 class MakerBase(ABC):
     @abstractmethod
-    def prepare_section(self): pass
+    def prepare_section(self, inner_coord: Location2D) -> None: pass
```
Comment (elsevers) [inline python/emach/tools/tool_abc/toolabc.py]: Indicate `Any` for return value

File: python/emach/tools/tool_abc/toolabc.py [hunk 5]
```diff
@@ -6,36 +6,42 @@
...
 
 # define abstract base class for MakerExtrudeBase
 class MakerExtrudeBase(MakerBase):
     @abstractmethod
-    def extrude(self): pass
+    def extrude(self, name: List[str], material: str, depth: float, token) -> int: pass
```
Comment (elsevers) [inline python/emach/tools/tool_abc/toolabc.py]: Indicate `Any` for return value

File: python/emach/tools/tool_abc/toolabc.py [hunk 6]
```diff
@@ -6,36 +6,42 @@
...
 
 # define abstract base class for MakerRevolveBase 
 class MakerRevolveBase(MakerBase):
     @abstractmethod
-    def revolve(self): pass
\ No newline at end of file
+    def revolve(self, name: List[str], material: str, center: Location2D,
```
Comment (elsevers) [inline python/emach/tools/tool_abc/toolabc.py]: Indicate `Any` for return value

**Review by elsevers — CHANGES_REQUESTED** [review]
See detailed comments. Need to create `TokenDraw` class.

**Review by aimthiazz — DISMISSED** [review]
Looks good - Indicated comments have been addressed.

**Review by ngadiyar93 — DISMISSED** [review]

**Review by mwjohnson56 — DISMISSED** [review]
All function signatures have been added to this class.

**Review by elsevers — APPROVED** [review]

**Review by ngadiyar93 — APPROVED** [review]

**Review by aimthiazz — APPROVED** [review]

**Pull request merged**